src/Identity/Core/src/IdentityBuilderExtensions.cs

```diff
@@ -17,9 +18,12 @@ public static class IdentityBuilderExtensions
     /// </summary>
     /// <param name="builder">The current <see cref="IdentityBuilder"/> instance.</param>
     /// <returns>The current <see cref="IdentityBuilder"/> instance.</returns>
+    [UnconditionalSuppressMessage("AOT", "IL3050", Justification = "MakeGenericType is safe because generic type and user type are reference types.")]
     public static IdentityBuilder AddDefaultTokenProviders(this IdentityBuilder builder)
     {
         var userType = builder.UserType;
+        Debug.Assert(userType.IsClass);
```

Comment:
This is public API, so the Debug.Assert is basically useless. Anyone can create one of these IdentityBuilder types and pass in any Type for the UserType.

We should make this an exception that happens all the time (AOTing or not) instead in order to be correct.